test/Microsoft.DotNet.ToolPackage.Tests/ToolPackageInstallerTests.cs

```diff
@@ -0,0 +1,639 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Transactions;
+using FluentAssertions;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Tools;
+using Microsoft.DotNet.Tools.Install.Tool;
+using Microsoft.DotNet.Tools.Tests.ComponentMocks;
+using Microsoft.Extensions.DependencyModel.Tests;
+using Microsoft.Extensions.EnvironmentAbstractions;
+using Xunit;
+
+namespace Microsoft.DotNet.ToolPackage.Tests
+{
+    public class ToolPackageInstallerTests : TestBase
```

Comment:
again, it is hard to find new tests, could you mark it?

Reply:
Tests added:

* `GivenSecondInstallWithoutATransactionTheFirstShouldNotRollback`
* `GivenAnInstalledPackageUninstallRemovesThePackage`
* `GivenAnInstalledPackageUninstallRollsbackWhenTransactionAborts`
* `GivenAnInstalledPackageUninstallRemovesThePackageWhenTransactionCommits`